app/lib/update_global_activity_analytics.rb

```diff
@@ -51,61 +21,106 @@ def perform
   private
 
   def fetch_data
```

Comment:
Since this method is only used in one place we could probably replace it with calling `google_analytics_report` directly.